src/sparseml/transformers/utils/initializers.py

```diff
@@ -186,6 +201,68 @@ def initialize_trainer(
     return trainer
 
 
+def resolve_recipe_application(
+    recipe: Union[str, Path, None], model_path: Union[str, Path]
+) -> Union[str, Path, None]:
+    """
+    Resolve the recipe to apply to the model.
+    If the recipe is None, will look for a recipe in the model_path
+
+    :param recipe: the recipe to apply to the model.
+        If None, will look for a recipe in the model_path
+    :param model_path: the path to the model to load
+    :return: the resolved recipe
+    """
```

Comment:
Nice, the logging messages are really clear, and good to have this in a helper function